config/initializers/devise.rb

```diff
@@ -205,10 +207,19 @@
   # Add a new OmniAuth provider. Check the wiki for more information on setting
   # up on your models and hooks.
   # config.omniauth :github, 'APP_ID', 'APP_SECRET', :scope => 'user,public_repo'
-  config.omniauth 'google_oauth2', Rails.application.secrets.google_key, Rails.application.secrets.google_secret, name: 'google'
-  config.omniauth :facebook, Rails.application.secrets.facebook_key, Rails.application.secrets.facebook_secret
-  config.omniauth :browser_id
-  config.omniauth :twitter, Rails.application.secrets.twitter_key, Rails.application.secrets.twitter_secret
+  config.omniauth 'google_oauth2', Rails.application.secrets.google_key, Rails.application.secrets.google_secret, name: 'google' if has_google?
+  config.omniauth :facebook, Rails.application.secrets.facebook_key, Rails.application.secrets.facebook_secret if has_facebook?
+  config.omniauth :browser_id if has_persona?
+  config.omniauth :twitter, Rails.application.secrets.twitter_key, Rails.application.secrets.twitter_secret if has_twitter?
+  config.omniauth :custom_oauth2, Rails.application.secrets.custom_oauth2_key, Rails.application.secrets.custom_oauth2_secret,
```

Comment:
Hey so I'm trying to figure out what happens when the day comes that we want to sign in with Github, and Dropbox, and Trello, and anyone else that has an OAuth signin. What are the challenges around adding support for N custom providers, instead of just the one?